src/Analysis/Ast/Impl/Values/Collections/PythonDictionary.cs

```diff
@@ -26,37 +26,37 @@ namespace Microsoft.Python.Analysis.Values.Collections {
     /// </summary>
     internal class PythonDictionary : PythonCollection, IPythonDictionary {
         private readonly Dictionary<IMember, IMember> _contents = new Dictionary<IMember, IMember>(new KeyComparer());
-        private readonly IPythonInterpreter _interpreter;
+        private readonly IPythonModule _declaringModule;
 
         public PythonDictionary(PythonDictionaryType dictType, IReadOnlyDictionary<IMember, IMember> contents, bool exact = false) :
             base(dictType, contents.Keys.ToArray(), exact: exact) {
             foreach (var kvp in contents) {
                 _contents[kvp.Key] = kvp.Value;
             }
-            _interpreter = dictType.DeclaringModule.Interpreter;
+            _declaringModule = dictType.DeclaringModule;
         }
 
-        public PythonDictionary(IPythonInterpreter interpreter, IMember contents, bool exact = false) :
-            base(new PythonDictionaryType(interpreter), Array.Empty<IMember>(), exact: exact) {
+        public PythonDictionary(IPythonModule declaringModule, IMember contents, bool exact = false) :
+            base(new PythonDictionaryType(declaringModule), Array.Empty<IMember>(), exact: exact) {
             if (contents is IPythonDictionary dict) {
                 foreach (var key in dict.Keys) {
                     _contents[key] = dict[key];
                 }
                 Contents = _contents.Keys.ToArray();
             }
-            _interpreter = interpreter;
+            _declaringModule = declaringModule;
         }
 
-        public PythonDictionary(IPythonInterpreter interpreter, IReadOnlyDictionary<IMember, IMember> contents, bool exact = false) :
-            this(new PythonDictionaryType(interpreter), contents, exact: exact) {
-            _interpreter = interpreter;
+        public PythonDictionary(IPythonModule declaringModule, IReadOnlyDictionary<IMember, IMember> contents, bool exact = false) :
+            this(new PythonDictionaryType(declaringModule), contents, exact: exact) {
+            _declaringModule = declaringModule;
```

Comment:
I don't see where we use `_declaringModule` directly. Looks like it can be still `_interpreter = declaringModule.Interpreter`

Reply:
OK